apertium/perceptron_spec.cc

```diff
@@ -476,17 +475,21 @@ PerceptronSpec::Machine::execCommonOp(Opcode op)
       stack.push(clamp(0, (int)untagged.size() - 1, stack.pop_off().intVal()));
       break;
     case GETWRD: {
-      //std::wcerr << "GETWRD start\n";
+      //std::cerr << "GETWRD start\n";
       stack.push(get_wordoid(tagged));
-      //std::wcerr << "GETWRD done\n";
+      //std::cerr << "GETWRD done\n";
     } break;
     case EXTOKSURF: {
-      std::wstring surf = get_token(untagged).TheSurfaceForm;
-      stack.push(new std::string(UtfConverter::toUtf8(surf)));
+      UString surf = get_token(untagged).TheSurfaceForm;
+      std::string temp;
+      utf8::utf16to8(surf.begin(), surf.end(), std::back_inserter(temp));
+      stack.push(temp);
```

Comment:
`std::move(temp)` and below. When done with a value that's going into a function or container, `move()` it away. But don't move returns.